syntax = "proto3";
package packets;

import "types.proto";

message EnterWorld {
	int64 ActorID = 1;
	int32 ZoneID = 2;
}

message EnterPacket {
    ActorType type = 1;
    int64 ActorID = 2;
    Vector3 Position = 3;
}

message Move {
    int64 ActorID = 1;
    Vector3 Position = 2;
}

message LeaveWorld {
}

message LeavePacket {
    int64 ActorID = 1;
}

message LoginReq {
    string ID = 1;
	string Pass = 2;
	int64 SerialID = 3;
}

message LoginAck {
	int64 UserID = 1;
	int64 ActorID = 2;
	ResultType Result = 3;
}

message CharSelectReq {
	int64 ClassType = 1;
}

message CharSelectAck {
	bool IsNewChar = 1;
	int64 UserID = 2;
	int64 ActorID = 3;
	int32 ZoneID = 4;
	ResultType Result = 5;
}

message ZoneMoveReq {
	int32 ZoneID = 1;
}

message ZoneMoveAck {
	ResultType Result = 1;
}


message PingReq {
}


message UpdateDispatchReq {
	message UpdateGold {
		message GoldData {
			int32 MoneyType = 1;
			int64 Add = 2;
			int64 Total = 3;
		}
		repeated GoldData GoldList = 1;
	}
	
	message UpdateItem {
		message ItemData {
			int64 SerialId = 1;
			int32 Durability = 2;
		}
		repeated ItemData AddList = 1;
		repeated ItemData DeleteList = 2;
		repeated ItemData UpdateList = 3;
	}
	
	int64 UserID = 1;
	repeated UpdateGold GoldList = 2;
	repeated UpdateItem ItemList = 3;
}

message VSDisconnectReq {
}

message VSDisconnectNotify {
}





